### PR TITLE
[timeseries] Fix incorrect scale of the quantile forecast for RecursiveTabular

### PR DESCRIPTION
**Pull request by shchur**
*Issue #, if available:*

*Description of changes:*
- Fix a regression caused by the new scaler logic in `RecursiveTabularModel` introduced in #4460
- Make `_in_sample_abs_seasonal_error` and `_in_sample_squared_seasonal_error` public methods

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py [hunk 1]
```diff
@@ -398,9 +397,14 @@ def _add_gaussian_quantiles(self, predictions: pd.DataFrame, repeated_item_ids:
         normal_scale_per_timestep = pd.Series(np.tile(sqrt_h, num_items), index=repeated_item_ids)
 
         residuals_std_per_timestep = self._residuals_std_per_item.reindex(repeated_item_ids)
-        # Use avg_residuals_std in case unseen item received for prediction
-        if residuals_std_per_timestep.isna().any():
-            residuals_std_per_timestep = residuals_std_per_timestep.fillna(value=self._avg_residuals_std)
```
Comment (shchur) [inline timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py:403]: After the refactor in #4460, we used the **scaled** `_avg_residuals_std` across all new items, which sometimes resulted in extreme prediction intervals for items not seen during training. For example, if items have very different scales, using the avg. scale across all items can result in too wide/too narrow intervals.

For example `cif_2016`, `prediction_length=12`, `eval_metric="SQL"`:
- AG v1.1.1
```
              model  score_test  score_val
0  RecursiveTabular    -1.01039   -0.86496
```
- Current `master`
```
              model    score_test  score_val
0  RecursiveTabular -1.140704e+06  -0.862594
```
- This PR
```
              model  score_test  score_val
0  RecursiveTabular   -1.008482  -0.862594
```

File: timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py [hunk 2]
```diff
@@ -398,9 +397,14 @@ def _add_gaussian_quantiles(self, predictions: pd.DataFrame, repeated_item_ids:
...
+        # Use in-sample seasonal error in for items not seen during fit
+        items_not_seen_during_fit = residuals_std_per_timestep.index[residuals_std_per_timestep.isna()].unique()
+        if len(items_not_seen_during_fit):
+            scale_for_new_items = np.sqrt(
```
Comment (abdulfatir) [inline timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py:403]: Will this give a single scale across all new items or 1 scale per unique item?

File: timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py [hunk 3]
```diff
@@ -398,9 +397,14 @@ def _add_gaussian_quantiles(self, predictions: pd.DataFrame, repeated_item_ids:
         normal_scale_per_timestep = pd.Series(np.tile(sqrt_h, num_items), index=repeated_item_ids)
 
         residuals_std_per_timestep = self._residuals_std_per_item.reindex(repeated_item_ids)
-        # Use avg_residuals_std in case unseen item received for prediction
-        if residuals_std_per_timestep.isna().any():
-            residuals_std_per_timestep = residuals_std_per_timestep.fillna(value=self._avg_residuals_std)
+        # Use in-sample seasonal error in for items not seen during fit
+        items_not_seen_during_fit = residuals_std_per_timestep.index[residuals_std_per_timestep.isna()].unique()
+        if len(items_not_seen_during_fit):
```
Comment (abdulfatir) [inline timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py:402]: nit: for better readability.
```suggestion
        if len(items_not_seen_during_fit) > 0:
```

File: timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py (hunk 2 above)
Comment (shchur) [reply timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py:403]: This returns a `pd.Series`, with `item_id` as index and the scale of the historic errors as value.

Added a type hint to make this more explicit.

**Review by abdulfatir — APPROVED** [review]

**Pull request merged**